parquet/src/test/java/org/apache/iceberg/parquet/TestParquetColumnIndex.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.parquet;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Function;
+import org.apache.avro.generic.GenericData;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.avro.AvroSchemaUtil;
+import org.apache.iceberg.data.parquet.GenericParquetReaders;
+import org.apache.iceberg.expressions.Expression;
+import org.apache.iceberg.expressions.Expressions;
+import org.apache.iceberg.io.CloseableIterator;
+import org.apache.iceberg.parquet.Parquet.ReadBuilder;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.apache.parquet.schema.MessageType;
+import org.apache.parquet.schema.PrimitiveType;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import static org.apache.iceberg.Files.localInput;
+import static org.apache.iceberg.TableProperties.PARQUET_PAGE_SIZE_BYTES;
+import static org.apache.iceberg.parquet.ParquetWritingTestUtils.writeRecords;
+import static org.apache.parquet.schema.PrimitiveType.PrimitiveTypeName.INT32;
+import static org.apache.parquet.schema.Type.Repetition.REQUIRED;
+
+public class TestParquetColumnIndex {
+
+  private MessageType parquetSchema = new MessageType("schema",
+      new PrimitiveType(REQUIRED, INT32, "intCol"));
+  private Schema schema = ParquetSchemaUtil.convert(parquetSchema);
+
+  @Rule
+  public TemporaryFolder temp = new TemporaryFolder();
+
+  @Test
+  public void testColumnIndexFilter() throws IOException {
+    File parquetFile = generateFileWithMultiplePages(ParquetAvroWriter::buildWriter);
+    int totalCount = getPageRecordCount(parquetFile, null);
+    int filterCount = getPageRecordCount(parquetFile,
+            Expressions.and(Expressions.notNull("intCol"), Expressions.equal("intCol", 1)));
+    Assert.assertTrue(filterCount < totalCount);
```

Comment:
Shouldn't this end up reading one and only one page? Can make the assertion more strict?
Based on my comment in https://github.com/apache/iceberg/pull/1566#discussion_r505847585 I doubt if the reduced row count is a result of page skipping or just row group skipping.

Reply:
Yes. I changed to test code to compare with one single page count.